### DISCUSSION TITLE
SimpleTheme: How to specify an exact background color for only an individual disabled ListBox?

### DISCUSSION THREAD

**Discussion by elgonzo** [AvaloniaUI/Avalonia#15677] [Q&A] [answered]
Using SimpleTheme (with Avalonia 11.1.0-beta2), how do i set a specific background color for a disabled listbox?

Note that this should not have an application-wide effect, as i want it to apply to only very specific ListBoxes. I want to use such ListBoxes to present groups of selectable images, and it is important that these ListBoxes maintain a black background in disabled state (there will be other indicators in the image items that indicate whether such an image item is responsive to clicks or not).

However, i fail to specify a background color for a disabled ListBox without the disabled ListBox altering the background color, and i don't know how to override this (apparently) in-built behavior.

The behavior appears to be related to the SimpleTheme, as the issue does not manifest when using FluentTheme. (Well, in both SimpleTheme and  FluentTheme, a similar behavior is present regarding altering the foreground color, but i am not concerned with foreground colors.)

Ideally, a robust solution for this should work independently from whatever theme is being used.

<br>

A very simple reproduction of a ListBox with a black background, and which is supposed to maintain this black background in disabled state:
```XML
<ListBox IsEnabled="False" Background="Black" Foreground="White" Margin="8" >
  <x:String>Foo</x:String>
  <x:String>Bar</x:String>
</ListBox>
```
But as can be seen in below screenshot, the ListBox alters the background color in disabled state, turning it into some dark/mid gray color:
![Black1](https://github.com/AvaloniaUI/Avalonia/assets/6129198/5c441b95-3c19-4f58-8bbe-35d382522d1d)

<br>

Well, okay, maybe i have to use styles for this, setting the black background color explicitly for the disabled state:
```XML
<ListBox  IsEnabled="False" Foreground="White" Margin="8">
	<ListBox.Styles>
		<Style Selector="ListBox">
			<Setter Property="Background" Value="Black" />
		</Style>
		<Style Selector="ListBox:disabled">
			<Setter Property="Background" Value="Black" />
		</Style>
	</ListBox.Styles>

	<x:String>Foo</x:String>
	<x:String>Bar</x:String>
</ListBox>
```
Nope, again no soup for me:
![Black2](https://github.com/AvaloniaUI/Avalonia/assets/6129198/838c8d89-4bf3-4788-89bb-1e5b8474c786)

I am able workaround this issue by making the background of the ListBox transparent and place the ListBox  into some Panel that takes on the job of realizing the background color for the ListBox, but doing that feels clumsy and not the cleanest way to achieve the desired outcome.

**Comment by jp2masa** [comment, marked as answer]
You have to change the opacity:

https://github.com/AvaloniaUI/Avalonia/blob/b25dc8ae80fcbb014a8cde00e871b5f4eddd424a/src/Avalonia.Themes.Simple/Controls/ListBox.xaml#L36-L38

**Reply by elgonzo** [reply]
Awesome! Replacing the `<Style Selector="ListBox:disabled">` with:
```XML
<Style Selector="ListBox:disabled /template/ Border#border">
    <Setter Property="Opacity" Value="1.0" />
</Style>
```
works. Thanks so much for your help!
